### ISSUE TITLE
Blocking other users

### ISSUE THREAD

**Issue by camelPhonso** [enBloc-org/kindly#99]
Users should be able to block other users from messaging them

**Comment by camelPhonso** [comment]
Still an issue to keep - will look for takers in unschool sessions

**Comment by ShaughnAnderson94** [comment]
We still want it we are just busy

**Comment by Angelica137** [comment]
!request

Can I pick this ticket?

**Comment by camelPhonso** [comment]
> Here is my rough idea of how to solve this issue. This can be either one big PR, or each step can be a separate ticket.. or we can also consolidate them into three tickets: step 1, steps 2-5, and step 6 ☺️

Thanks @nataliiazab 😃 - I think breaking it down is probably best. And it looks like you've got a good idea of how smaller tickets could be divided so I would go ahead with submitting those (once we decide to close this `spike` ticket) 👍 

> - To adjust the messaging logic (e.g. in `MessageForm.tsx` file) to check if the user is blocked by the recipient before allowing the message to be sent.
> - If the user is blocked - add a message 'You cannot send messages to this user as you have been blocked' and prevent the message from being sent
> - To ensure that messages from blocked users are not displayed - in the CurrentConversation component,

We'd probably want to just stop people from interacting with users that have blocked them altogether. So instead of receiving a notification that they cannot message someone they would just not be able to view that user's items or something 🤔 

This would be possible with the schema changes you propose but would require some serious thinking about how we are fetching/filtering items. We might want to set up a planning session and create some diagrams of the flow of this feature to then pass to Kat so it can be integrated in the new design.

This aspect of the feature needs to be settled with some more planning that will probably require the core team to get together but your general plan is looking great - I think the schema your proposing will do quite well and I agree with your idea to divide this into three smaller tickets 👍

**Comment by nataliiazab** [comment]
Thank you for your comment! I absolutely agree that we should block users from viewing others' items 

Here are a few thoughts from our meeting last week, leaving them here just for the future reference:
- It would be good to schedule a separate meeting to clarify what actions should be taken for this ticket and how to divide the tasks.
- Katia mentioned that we could consider implementing a "report user" feature first, in case blocking users takes too much time. For example, users could report others to the admins of Kindly, which might trigger an automatic email to the admins. This way, the admins could restrict access for the reported user across the entire platform.

**Comment by nataliiazab** [comment]
These are just references of what can be added, I can explain more when we have a planning session

![Image](https://github.com/user-attachments/assets/bb1a4c07-964b-4296-aba1-03c28eaff3f3)
![Image](https://github.com/user-attachments/assets/60bf458b-9ba5-4fa2-b48e-89a13a4bc519)
![Image](https://github.com/user-attachments/assets/5669dce6-810c-4a6d-9b74-346bddca4d1f)
